packages/firestore/src/core/query.ts

```diff
@@ -631,6 +643,24 @@ export class KeyFieldFilter extends FieldFilter {
   }
 }
 
+/** Filter that matches on key fields within an array. */
+export class KeyFieldInFilter extends FieldFilter {
```

Comment:
On the surface it's bugging me a little that we have `KeyFieldFilter`, `InFilter` and `KeyFieldInFilter` as peers... though thinking about this, I think it probably makes sense.  cc/ @wilhuff to sanity-check since he just put thought into our filter hierarchy...

Reply:
Abstractly, `KeyFieldInFilter` seems like it is-a `KeyFieldFilter` and `KeyFieldInFilter` seems like it is-a `InFilter`, but it's also neither of them because it operates on different arguments than either.

That is to say, none of these are practically substitutable for each other and subclasses that violate LSP seem like a bad idea.

Reply:
👍 Thanks for the sanity check.